nixos/modules/installer/cd-dvd/iso-image.nix

```diff
@@ -53,6 +53,8 @@ let
       image = "/boot/${config.system.boot.loader.kernelFile}";
       initrd = "/boot/initrd";
     };
+
+  targetArch = (config.boot.loader.grub.forcei686 ? "ia32") or stdenv.hostPlatform.efiArch;
```

Comment:
I'm not sure this does what you think it does?  The `?` operator tests for the presence of the name of the right hand side in an attribute set on the left-hand side — e.g. `{ foo = 1; } ? foo` is `true`.

I'd expect to see something like:

```suggestion
  targetArch =
    if config.boot.loader.grub.forcei686 then
      "ia32"
    else
      stdenv.hostPlatform.efiArch;

```

Reply:
Hmm, interesting... I tested that and it has worked... But your option is more "readable", so let's use it.